cli_tools/gce_windows_upgrade/upgrader/upgrader.go

```diff
@@ -139,9 +148,60 @@ func (u *upgrader) init() error {
 	if err != nil {
 		return daisy.Errf("Failed to create GCE client: %v", err)
 	}
+
+	err = u.initLogger()
+	if err != nil {
+		return daisy.Errf("Failed to initialize logger: %v", err)
+	}
+
 	return nil
 }
 
+func (u *upgrader) initLogger() error {
+	if u.CloudLogsDisabled || u.ProjectPtr != nil {
+		return nil
+	}
+
+	loggingOptions := []option.ClientOption{option.WithCredentialsFile(u.Oauth)}
+	cloudLoggingClient, err := logging.NewClient(u.ctx, *u.ProjectPtr, loggingOptions...)
+	if err != nil {
+		return err
+	}
+
+	if err := cloudLoggingClient.Ping(u.ctx); err != nil {
+		fmt.Println("Failed to initialize cloud logger: ", err)
+	} else {
+		cloudLogName := fmt.Sprintf("windows-upgrade-%s", u.id)
+		u.cloudLogger = cloudLoggingClient.Logger(cloudLogName)
+		go func() {
+			for {
+				time.Sleep(5 * time.Second)
+				u.cloudLogger.Flush()
+			}
+		}()
```

Comment:
consider extracting it to a separate func, similarly to what Daisy does, like flushLog() or similar

Reply:
done